src/stringtable.rs

```diff
@@ -223,14 +223,14 @@ impl<'st> StringRef<'st> {
 }
 
 /// Read-only version of the string table
-pub struct StringTable<'data> {
+pub struct StringTable {
     // TODO: Replace with something lazy
-    string_data: &'data [u8],
+    string_data: Vec<u8>,
     index: FxHashMap<StringId, Addr>,
 }
 
-impl<'data> StringTable<'data> {
-    pub fn new(string_data: &'data [u8], index_data: &[u8]) -> StringTable<'data> {
+impl<'data> StringTable {
+    pub fn new(string_data: Vec<u8>, index_data: Vec<u8>) -> StringTable {
```

Comment:
`index_data` can stay a slice since we only construct the `index` table from it but don't hold on to it.